sync/src/types.rs

```diff
@@ -836,6 +868,16 @@ impl SyncState {
     pub fn get_orphan_block(&self, block_hash: &Byte32) -> Option<core::BlockView> {
         self.orphan_block_pool.get_block(block_hash)
     }
+
+    pub fn insert_peer_unknown_header_list(&self, pi: PeerIndex, header_list: Vec<Byte32>) {
```

Comment:
It's more common to pass slice as arg rather than vector when you just want to provide read-only access

```suggestion
    pub fn insert_peer_unknown_header_list(&self, pi: PeerIndex, header_list: &[Byte32]) {
```
then we can remove the mut modifier `block_locator_hashes` from `get_headers_process.rs`

```
insert_peer_unknown_header_list(self.peer, &block_locator_hashes[1..])
```

Reply:
types.rs#877 it need ownership